### PR TITLE
C: Include only required headers

### PR DESCRIPTION
**Pull request by Smit-create**
#556

### CODE REVIEW

File: src/libasr/codegen/asr_to_c_cpp.h [hunk 1]
```diff
@@ -561,43 +563,6 @@ R"(#include <stdio.h>
...
-            }
-            case (ASR::cast_kindType::LogicalToInteger) : {
-                src = "(int)(" + src + ")";
-                break;
-            }
-            default : throw CodeGenError("Cast kind " + std::to_string(x.m_kind) + " not implemented");
-        }
-        last_expr_precedence = 2;
-    }
-
```
Comment (certik) [inline src/libasr/codegen/asr_to_c_cpp.h:600]: Why did you remove these? 

Is it needed for this PR? If not, let's do it separately. I am hoping to have as much in the common visitor as possible, but it looks like currently they are overridden in the subclasses (C and C++ visitors)?

**Comment by certik** [comment]
This is a great idea, thanks for doing it. We can later put the ASSERT macros into our runtime library header file, and also only include the runtime library if it is being used.

In general, I want both the C and C++ backend to eventually produce code that you would write by hand if you translated Python or Fortran into C or C++ manually. So this change goes in that direction.

File: src/libasr/codegen/asr_to_c_cpp.h (hunk 1 above)
Comment (Smit-create) [reply src/libasr/codegen/asr_to_c_cpp.h:600]: Okay, will change that in new PR.

**Comment by certik** [comment]
I think that this is good, thanks! Can you please resolve conflicts?

**Review by certik — APPROVED** [review]
I think this loos good, thanks!
